### PR TITLE
Fill up dg

### PR DESCRIPTION
**Pull request by guangxuan**
Adds-
User stories
Use cases
a bit more NFRs
Manual testing
Effort

Published here: https://guangxuan.github.io/tp/DeveloperGuide.html

How much should the existing diagrams be changed?

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -527,40 +656,149 @@ testers are expected to do more *exploratory* testing.
...
+add buyer n/tom p/91234567 e/sam@email.com t/condo t/3rm $/10000
+match    auto
+clear
+```
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
+   * Make sure the `PropertyWhiz` is currently not running.
+   * Open the `addressbook.json` file in the /data folder with your favourite text editor.
```
Comment (sunjc826) [inline docs/DeveloperGuide.md:775]: `propertywhiz.json`

**Review by sunjc826 — COMMENTED** [review]
TODOs: (The rest of us can work on this as well)
- MatchCard and MatchListPanel need to be added to "UI Component"
- ModelComponent is outdated now due to double lists (currList and fullList)
-

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -451,65 +452,193 @@ Priorities: High (must have) - `H`, Medium (nice to have) - `M`, Low (unlikely t
...
+
+1. `User` requests to find properties by specifying keywords and a set of tags.
+2.  `PropertyWhiz` filters the current property list to include only properties that contain the specified keywords in the name and have the specified tags.
+
+Use case ends.
+
+**Use case: UC06 - Import properties**
+1. `User` requests to import properties.
+2. `User` chooses the source csv file.
+3. `PropertyWhiz` imports the rows within the csv file as properties to the current property list.
```
[the remark this answers is not in the document]
Comment (guangxuan) [reply docs/DeveloperGuide.md:552]: Import prepends the newly added list to the current property list

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -451,65 +452,193 @@ Priorities: High (must have) - `H`, Medium (nice to have) - `M`, Low (unlikely t
...
+      Use case resumes at step 2.
+
+**Use case: UC09 - Stat Properties**
+1. `User` requests to draw a histogram of property prices.
+2. `PropertyWhiz` draws and shows in a new window, a histogram of property prices within the current property list. 
+
+**Extensions**
+
+* 2a. There are no properties in the current property list.
+    * 2a1. `PropertyWhiz` shows an error message.
```
[the remark this answers is not in the document]
Comment (guangxuan) [reply docs/DeveloperGuide.md:617]: Yes, the result behaves like an error, i.e. command changes colour to red.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -451,65 +452,193 @@ Priorities: High (must have) - `H`, Medium (nice to have) - `M`, Low (unlikely t
...
+      Use case ends.
+
+* 2b. There is already a window containing a previously drawn histogram.
+    * 2a1. `PropertyWhiz` closes the window containing the previously drawn histogram.
+
+      Use case resumes at step 2.
 
 ### Non-Functional Requirements
 
 1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
```
[the remark this answers is not in the document]
Comment (guangxuan) [reply docs/DeveloperGuide.md:628]: Not sure tbh

**Review by sunjc826 — APPROVED** [review]
Lgtm

**Pull request merged**